app/models/project_duplicator.rb

```diff
@@ -1,15 +1,36 @@
 class ProjectDuplicator
-  attr_reader :project_to_copy
+  attr_reader :orig, :copy
 
-  def initialize(project_to_copy)
-    @project_to_copy = project_to_copy
+  def initialize(orig)
+    @orig = orig
   end
 
   def duplicate
-    copy = project_to_copy.amoeba_dup
-    copy.original = project_to_copy
+    @copy = orig.amoeba_dup
+    copy.original = orig
+
+    duplicate_timeline_entry(orig.root_timeline_entry)
 
     copy.save!
     copy
   end
+
+  # make sure to set project_id on copy
```

Comment:
This comment is not really accurate anymore. Perhaps "Recursively copies timeline entries in pre-order traversal, ensuring correct parentage."